src/main/java/uk/gov/justice/digital/zone/Zone.java

```diff
@@ -53,4 +54,9 @@ protected void appendToDeltaLakeTable(Dataset<Row> dataFrame, String tablePath)
         logger.info("Append completed successfully");
     }
 
+    protected Dataset<Row> createEmptyDataFrame(Dataset<Row> dataFrame) {
+        return dataFrame.filter(lit(false));
```

Comment:
Why not explicitly create an empty dataframe with the correct schema instead? E.g.

```
spark.createDataFrame(spark.emptyDataFrame().javaRDD(), dataFrame.schema());
```

Reply:
nice one